### PR TITLE
Skip upgrade step when plan is already selected

### LINKED ISSUE #93300: Handle case when user already selected a plan before coming to the flow
### Details

When the user already selected a plan before coming to the flow (currently it would happen on the `/move` landing page), we should directly create the site and redirect them to the checkout. Notice that in this case, we don't have the plan step in the middle of the site creating and checkout.

### Checklist

- [ ] The flow is prepared to start also when selecting a plan before accessing it.

### Related

https://github.com/Automattic/wp-calypso/issues/93258

### PR DESCRIPTION
**Pull request by renatho**
Related to #93300 

## Proposed Changes

* Handle cases coming to the migration flow with a plan already defined.
* Remove step `import-or-migrate-with-upgrade` which wasn't used.

### Alternative solution

I started implementing it in a way that the Upgrade Plan would be responsible for submitting automatically when the plan was already defined, but after thinking more, I decided to handle it directly in the flow, where it seems to make more sense.

## Why are these changes being made?

* We will have a case where the users will come from a landing page already selecting a plan.

## Testing Instructions

* Navigate to `/setup/migration/create-site?platform=wordpress&plan=business`, and make sure a site is created and you are redirected directly to the checkout.
* Navigate to `/setup/migration/create-site?platform=wordpress&plan=unknownplan` and make sure you are redirected to the upgrade plan step.
* Navigate through the flow `/setup/migration`, and make sure it continues working properly.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/data/site-migration/use-checkout-migration-introductory-offer-sticker.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const useCheckoutMigrationIntroductoryOfferSticker = (
 	const urlQueryParams = new URLSearchParams( window.location.search );
 	const refPath = urlQueryParams?.get( 'ref' ) ?? '';
 	const checkoutBackPath = urlQueryParams?.get( 'checkoutBackUrl' ) ?? '';
+	const introductoryOffer = urlQueryParams?.get( 'introductoryOffer' ) ?? '';
```
Comment (renatho) [inline client/data/site-migration/use-checkout-migration-introductory-offer-sticker.ts:13]: This hook is modified to also add the introductory offer directly in the checkout when the `introductoryOffer` is set.

File: client/landing/stepper/declarative-flow/migration/index.tsx [hunk 2]
```diff
@@ -29,14 +34,68 @@ const {
...
 const useCreateStepHandlers = ( navigate: Navigate< StepperStep[] >, flowObject: Flow ) => {
 	const [ query ] = useSearchParams();
+	const flowPath = ( flowObject.variantSlug ?? flowObject.name ) as string;
 
 	const getFromPropsOrUrl = ( key: string, props?: ProvidedDependencies ): Primitive => {
 		const value = props?.[ key ] || query.get( key );
 		return typeof value === 'object' ? undefined : ( value as Primitive );
 	};
 
+	const navigateToCheckout = ( {
```
Comment (renatho) [inline client/landing/stepper/declarative-flow/migration/index.tsx:52]: Extracted to a separate function since it's used in 2 different steps.

**Review by gabrielcaires — APPROVED** [review]
LGTM and thanks for the automated tests.
They are very helpful to understand the changes

**Comment by renatho** [comment]
The changes post-approval here were related to a rebase only.

**Pull request merged**